packages/storybook/src/stories/template-generator.ts

```diff
@@ -0,0 +1,107 @@
+import { Story } from '@storybook/web-components';
+import docsJson from '../../elements-stencil-docs';
+
+/**
+ * A utility class that helps with creating type-safe stories
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+export class TemplateGenerator<Component extends Object> {
+  private readonly tagName: string;
+  private readonly templateFn: Story<Component>;
+
+  constructor(tagName: string, templateFn: Story<Component>) {
+    this.tagName = tagName;
+    this.templateFn = templateFn;
+  }
+
+  /**
+   * Generates the default template story.
+   */
+  public generatePlaygroundStory() {
+    return this.generateTemplate();
+  }
+
+  /**
+   * Generates a story based of a key-value-pair and automatically adds the documentation of the property.
+   *
+   * @param key The key of the property
+   * @param value The value of the property which is assignable to `key`
+   */
+  public generateStory<Key extends keyof Component>(
+    key: Key,
+    value: Component[Key]
+  ) {
+    const argument: Component = ({ [key]: value } as unknown) as Component;
+
+    return this.generateTemplate(argument, key);
+  }
+
+  /**
+   * Generates a new story based of multiple arguments.
+   * Adds a description of the parameter which is provided by `prop`.
+   *
+   * @param args Properties to assign to the template
+   * @param prop The name of the "main" property the stories is written for
+   */
+  public generateStoryWithArgs(args: Component, prop: keyof Component) {
```

Comment:
private?

Reply:
That's one other way to generate a story. IMO there are three different scenarios:
1. Create a playground story (`generatePlaygroundStory`)
2. Create a story for a single property where no other properties have to be set (`generateStory(...)` e.g.: `generateStory('checked', true)`) 
3. Create a more complex story where you have to set multiple properties at once ( `generateStoryWithArgs`), e.g.: 
```ts
generateStoryWithArgs(
	{ 
		checked: true,
		selection: true,
		disabled: true,
	},
	'disabled'
)
```